src/main/java/core/basesyntax/SquareSupplier.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class SquareSupplier {
+    private Random random = new Random();
+    private RandomSize size = new RandomSize();
+    private ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomSquare() {
+        return new Square(size.getRandomSize(), color.getRandomColor());

```

Comment:
'Square' class should have two parameters, 'color' and 'side'. It seems you are passing size as color parameter and color as size, which is incorrect.

Reply:
I guess not. My constructor had a size in first position. Now after implementing your comment about hierarchy in constructor I changed it and recoded also this order.

Reply:
I guess not. My constructor has a size in first position.